### ISSUE TITLE
Issue while renaming doce output files

### ISSUE THREAD

**Issue by modantailleur** [mathieulagrange/doce#211]
I've been facing an issue with doce when renaming my data.

Let's say for example that I have 4 file names,
classifier=PANN+dataset=full+dilation=0+epoch=10+kernel_size=3+learning_rate=-3+nb_channels=16+nb_layers=1+step=train+transcoder=mlp_pinv_loss.npy
classifier=PANN+dataset=full+dilation=0+epoch=10+kernel_size=3+learning_rate=-3+nb_channels=16+nb_layers=1+step=train+transcoder=mlp_loss.npy
classifier=PANN+dataset=full+dilation=0+epoch=10+kernel_size=3+learning_rate=-3+nb_channels=16+nb_layers=1+step=train+transcoder=mlp_pinv_loss_eval.npy
classifier=PANN+dataset=full+dilation=0+epoch=10+kernel_size=3+learning_rate=-3+nb_channels=16+nb_layers=1+step=train+transcoder=mlp_loss_eval.npy

I want to rename all those files to add "ts=0" in the name, so that they would look like this:
classifier=PANN+dataset=full+dilation=0+epoch=10+kernel_size=3+learning_rate=-3+nb_channels=16+nb_layers=1+step=train+transcoder=mlp_pinv+ts=0_loss.npy
classifier=PANN+dataset=full+dilation=0+epoch=10+kernel_size=3+learning_rate=-3+nb_channels=16+nb_layers=1+step=train+transcoder=mlp+ts=0_loss.npy
classifier=PANN+dataset=full+dilation=0+epoch=10+kernel_size=3+learning_rate=-3+nb_channels=16+nb_layers=1+step=train+transcoder=mlp_pinv+ts=0_loss_eval.npy
classifier=PANN+dataset=full+dilation=0+epoch=10+kernel_size=3+learning_rate=-3+nb_channels=16+nb_layers=1+step=train+transcoder=mlp+ts=0_loss_eval.npy

As you can see, with what's currently written on the documentation  in the section "removing a factor", this is not easily feasable. I can't just find the last "_" sign and put the "ts=0" at this position, since some metrics are named "loss" and others "loss_eval". I can't find "transcoder=mlp", because it will add a ts=0 between "mlp" and "pinv".

A solution would be to add three functions:
- add factor: take as input the name of the factor to add, the list of outputs names (in my example "loss" and "loss_eval") and the default value I want to give to every existing file for this factor.
- remove factor: take as input the name of the factor to remove, the list of output names, and the default value to keep (will remove every existing file that doesn't match this value)
- rename factor: take as input the name of the factor to rename

**Comment by mathieulagrange** [comment]
Would it be of interested to change the "_" to something like : 
 
 >
 ->
 -->

What is your opinion ?

**Comment by mathieulagrange** [comment]
hello, you can now explicitly set your delimiter (see demo.py). Can you play around with this and let me know ?
